PyDSS/pydss_results.py

```diff
@@ -183,21 +226,52 @@ def export_data(self, path=None, fmt="csv", compress=False):
 
         for elem_class in self.list_element_classes():
             for prop in self.list_element_properties(elem_class):
-                df = self.get_full_dataframe(elem_class, prop)
+                try:
+                    df = self.get_full_dataframe(elem_class, prop)
+                except InvalidParameter:
+                    logger.info(f"cannot create full dataframe for %s %s", elem_class, prop)
+                    self._export_filtered_dataframes(elem_class, prop, path, fmt, compress)
+                    continue
                 base = "__".join([elem_class, prop])
                 filename = os.path.join(path, base + "." + fmt.replace(".", ""))
                 write_dataframe(df, filename, compress=compress)
 
+        if self._elem_prop_sums:
+            data = copy.deepcopy(self._elem_prop_sums)
+            for elem_class, prop, name, val in self.iterate_element_property_sums():
+                # JSON lib cannot serialize complex numbers.
+                if isinstance(val, np.ndarray):
+                    new_val = []
+                    convert_str = val.dtype == "complex"
+                    for item in val:
+                        if convert_str:
+                            item = str(item)
+                        new_val.append(item)
+                    data[elem_class][prop][name] = new_val
+                elif isinstance(val, complex):
+                    data[elem_class][prop][name] = str(val)
+
+            filename = os.path.join(path, "element_property_sums.json")
+            dump_data(data, filename, indent=2)
+
         logger.info("Exported data to %s", path)
 
-    def get_dataframe(self, element_class, prop, element_name, **kwargs):
+    def _export_filtered_dataframes(self, elem_class, prop, path, fmt, compress):
+        for name, df in self.iterate_dataframes(elem_class, prop):
+            base = "__".join([elem_class, prop, name])
+            filename = os.path.join(path, base + "." + fmt.replace(".", ""))
+            write_dataframe(df, filename, compress=compress)
+
+    def get_dataframe(self, element_class, prop, element_name, real_only=False, **kwargs):
         """Return the dataframe for an element.
 
         Parameters
         ----------
         element_class : str
         prop : str
         element_name : str
+        real_only : bool
+            If dtype of any column is complex, drop the imaginary component.
```

Comment:
@AadilLatif I added this because in some situations I've been told that people only want the real component of a complex number.  If you disagree with having this, I'll take it out.

Reply:
I am fine with it